### PR TITLE
feat: add MVI control methods

### PR DESCRIPTION
**Pull request by nand4011**
Create initial MVI client with the create, list, and delete control methods.

Create initial configuration objects for the vector index client.

Generify the names of the simplest trace logging methods.

Fix a casting bug in the auth and token configuration equals methods.

### CODE REVIEW

File: src/Momento.Sdk/Config/Transport/StaticVectorIndexTransportStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /// <param name="obj"></param>
+    /// <returns></returns>
+    public override bool Equals(object obj)
+    {
+        if ((obj == null) || !this.GetType().Equals(obj.GetType()))
+        {
+            return false;
+        }
+
+        var other = (StaticTransportStrategy)obj;
```
Comment (malandis) [inline src/Momento.Sdk/Config/Transport/StaticVectorIndexTransportStrategy.cs:52]: StaticVectorIndexTransportStrategy?

File: src/Momento.Sdk/Config/VectorIndexConfigurations.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            );
+            return new Laptop(finalLoggerFactory, transportStrategy);
+        }
+    }
+
+    /// <summary>
+    /// InRegion provides defaults suitable for an environment where your client is running in the same region as the Momento
+    /// service.  It has more agressive timeouts and retry behavior than the Laptop config.
+    /// </summary>
+    public class InRegion
```
Comment (malandis) [inline src/Momento.Sdk/Config/VectorIndexConfigurations.cs:63]: We can remove this for now

File: src/Momento.Sdk/IPreviewVectorIndexClient.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    /// the following subtypes:
+    /// <list type="bullet">
+    /// <item><description>CreateVectorIndexResponse.Success</description></item>
+    /// <item><description>CreateVectorIndexResponse.IndexAlreadyExists</description></item>
+    /// <item><description>CreateVectorIndexResponse.Error</description></item>
+    /// </list>
+    /// Pattern matching can be used to operate on the appropriate subtype.
+    /// For example:
+    /// <code>
+    /// if (response is CreateCacheResponse.Error errorResponse)
```
Comment (malandis) [inline src/Momento.Sdk/IPreviewVectorIndexClient.cs:36]:
```suggestion
    /// if (response is CreateVectorIndexResponse.Error errorResponse)
```

File: src/Momento.Sdk/PreviewVectorIndexClient.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+using System.Threading.Tasks;
+using Momento.Sdk.Auth;
+using Momento.Sdk.Config;
+using Momento.Sdk.Internal;
+using Momento.Sdk.Requests.Vector;
+using Momento.Sdk.Responses.Vector;
+
+namespace Momento.Sdk;
+
+public class PreviewVectorIndexClient: IPreviewVectorIndexClient
```
Comment (malandis) [inline src/Momento.Sdk/PreviewVectorIndexClient.cs:10]: Can you add a docstring for the class too?

File: src/Momento.Sdk/Responses/Vector/CreateVectorIndexResponse.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+{
+
+    /// <include file="../../docs.xml" path='docs/class[@name="Success"]/description/*' />
+    public class Success : CreateVectorIndexResponse { }
+
+    /// <summary>
+    /// Class <c>IndexAlreadyExists</c> indicates that an index with the requested name
+    /// has already been created in the requesting account.
+    /// </summary>
+    public class IndexAlreadyExists : CreateVectorIndexResponse { }
```
Comment (malandis) [inline src/Momento.Sdk/Responses/Vector/CreateVectorIndexResponse.cs:41]: We use `AlreadyExists` elsewhere. .NET and Python are outliers. Let's correct there.

File: src/Momento.Sdk/IPreviewVectorIndexClient.cs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    /// <item><description>
+    /// Euclidean similarity, the sum of squared differences, is appropriate for datasets where
+    /// this metric is meaningful. For example, if the vectors represent images, and the embedding
+    /// model is trained to optimize the euclidean distance between images, then euclidean
+    /// similarity is appropriate.
+    /// </description></item>
+    /// </list>
+    /// </remarks>
+    /// </returns>
+    public Task<CreateVectorIndexResponse> CreateIndexAsync(string indexName, ulong numDimensions, SimilarityMetric similarityMetric = SimilarityMetric.CosineSimilarity);
```
Comment (malandis) [inline src/Momento.Sdk/IPreviewVectorIndexClient.cs:60]:
```suggestion
    public Task<CreateVectorIndexResponse> CreateIndexAsync(string indexName, long numDimensions, SimilarityMetric similarityMetric = SimilarityMetric.CosineSimilarity);
```

Should be a `long` to be CLS compliant

Comment (nand4011) [reply src/Momento.Sdk/IPreviewVectorIndexClient.cs:60]: I hadn't heard or CLS compliance. I'll change it.

**Review by malandis — APPROVED** [review]

**Pull request merged**